src/main/java/de/rwth/idsg/steve/repository/impl/GenericRepositoryImpl.java

```diff
@@ -52,6 +59,25 @@ public class GenericRepositoryImpl implements GenericRepository {
 
     @Autowired private DSLContext ctx;
 
+    @EventListener
```

Comment:
It is mixing repository and service layer.

Why not move the logic into a service and use a `getMySQLOffsets()` provided by the repository?